### PR TITLE
Client Functional Standup

### PR DESCRIPTION
**Pull request by zwrose**
This PR includes a functionally-complete (but NOT at all styled) React client for the our desired initial functionality.

Error handling is included, at least at a basic level.

Fetch IS implemented, even though the server is not yet up - errors are expected, and there is a graceful fallback.

### CODE REVIEW

File: client/src/hooks/useUpdateServer.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { useUpdateEffect } from 'react-use';
+
+export default function useUpdateServer(updateData, updateURL, setErrorMsgs, deps) {
```
Comment (Ryan-Rose-Bose) [inline client/src/hooks/useUpdateServer.js:3]: Consider moving the base URL logic into this function and only providing an endpoint as an argument.

File: client/src/components/App.js [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+import Grid from '@material-ui/core/Grid';
+import CssBaseline from '@material-ui/core/CssBaseline';
+import LightsOnOff from 'components/LightsOnOff';
+import GrillLightsControls from 'components/GrillLightsControls';
+import Backdrop from '@material-ui/core/Backdrop';
+import CircularProgress from '@material-ui/core/CircularProgress';
+import ErrorBlock from 'components/ErrorBlock'
+
+export default function App() {
+  // constants //
```
[the remark this answers is not in the document]
Comment (zwrose) [reply client/src/components/App.js:13]: bailed on the trailing slashes, was being too fancy

File: client/src/components/App.js [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+  const SERVER_URL = useRef(window.location + 'state');
```
[the remark this answers is not in the document]
Comment (zwrose) [reply client/src/components/App.js:14]: leaving, see our offline convos :)

File: client/src/components/App.js [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+  // initialize children state from the server //
+  const lightsOnOffInitial = useRef({});
+  const grillLightsControlsInitial = useRef({});
+  
+  useEffect(() => {
+    // fetch initial server state
+    fetch(SERVER_URL, {
+      headers: {
+        'Content-Type': 'application/json',
```
[the remark this answers is not in the document]
Comment (zwrose) [reply client/src/components/App.js:42]: removed

File: client/src/components/App.js [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+      },
+    })
+    .then(response => {
```
[the remark this answers is not in the document]
Comment (zwrose) [reply client/src/components/App.js:45]: could in the future, but leaving this for now

File: client/src/components/GrillLightsControls.js [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+import ColorSpeedConfig from 'components/ColorSpeedConfig';
+import RacerConfig from 'components/RacerConfig';
+import RainbowConfig from 'components/RainbowConfig';
+import useUpdateServer from 'hooks/useUpdateServer';
+
+export default function GrillLightsControls(props) {
+  // constants //
+  const DEFAULT_COLORS = {r: props.initialState.red, g: props.initialState.green, b: props.initialState.blue};
+  const SAMPLE_DELAY = 250;
+  const SERVER_URL = window.location + 'animation';
```
[the remark this answers is not in the document]
Comment (zwrose) [reply client/src/components/GrillLightsControls.js:17]: updated

File: client/src/hooks/useUpdateServer.js (hunk 1 above)
Comment (zwrose) [reply client/src/hooks/useUpdateServer.js:3]: I'd rather leave this hook a bit more generic, any harm in leaving as is?

Comment (rytrose) [reply client/src/hooks/useUpdateServer.js:3]: No harm, but I still recommend it at a later date!

**Pull request merged**